### PR TITLE
APPSRE-6644 Refactor get_updated_recommended_versions add channels and initial version

### PR DESCRIPTION
**Pull request by janboll**
Refactor get_updated_recommended_versions to add channel to the recommendedVersion. Also removed the possibility of having manual settings for recommendedVersions. Recommendedversion are now generated only, it was never a requirement to able to set them manually.

Stil missing: initial version

Required schema change: https://github.com/app-sre/qontract-schemas/pull/349

### CODE REVIEW

File: reconcile/ocm_update_recommended_version.py [hunk 1]
```diff
@@ -58,42 +58,45 @@ def get_version_weights(ocm: dict[str, Any]) -> tuple[int, int]:
...
             )
-            rv_updated.append(rv_current)
-        elif len(rv_workload) == 1 and len(versions) == 0:
-            # No clusters exist, but a rv was added, probably a manual setting
-            rv_updated.append(rv_workload[0])
+
+    for cwv_items in channel_workload_versions.items():
+        cwv, versions = cwv_items
+        if len(cwv.split("/")) != 2:
+            raise ValueError("Expecting workload/channel format!, got: {}".format(cwv))
```
Comment (chassing) [inline reconcile/ocm_update_recommended_version.py:89]: Nit: Please use an f-string here.

**Review by chassing — APPROVED** [review]
LGTM

File: reconcile/ocm_update_recommended_version.py (hunk 1 above)
Comment (janboll) [reply reconcile/ocm_update_recommended_version.py:89]: copilot did that ;)

Comment (chassing) [reply reconcile/ocm_update_recommended_version.py]: LOL

File: reconcile/ocm_update_recommended_version.py [hunk 2]
```diff
@@ -58,42 +58,45 @@ def get_version_weights(ocm: dict[str, Any]) -> tuple[int, int]:
...
-                versions, high_weight, majority_weight
+
+    channel_workload_versions: dict[str, list[str]] = {}
+
+    for uc in ocm_info["upgradePolicyClusters"] or []:
+        cluster_name = uc["name"]
+        for workload in uc["upgradePolicy"]["workloads"]:
+            channel_workload = f"{workload}/{cluster[cluster_name].spec.channel}"
+            if channel_workload not in channel_workload_versions:
+                channel_workload_versions[channel_workload] = []
```
Comment (patjlm) [inline reconcile/ocm_update_recommended_version.py:81]: really nit: `channel_workload_versions.setdefault(channel_workload, [])`

File: reconcile/ocm_update_recommended_version.py [hunk 3]
```diff
@@ -58,42 +58,45 @@ def get_version_weights(ocm: dict[str, Any]) -> tuple[int, int]:
...
-            # Managed clusters exist for this workload
-            rv_current["recommendedVersion"] = recommended_version(
-                versions, high_weight, majority_weight
+
+    channel_workload_versions: dict[str, list[str]] = {}
+
+    for uc in ocm_info["upgradePolicyClusters"] or []:
+        cluster_name = uc["name"]
+        for workload in uc["upgradePolicy"]["workloads"]:
+            channel_workload = f"{workload}/{cluster[cluster_name].spec.channel}"
```
Comment (patjlm) [inline reconcile/ocm_update_recommended_version.py:79]: why not using a tuple ?

**Pull request merged**